cwd/editor.js

```diff
@@ -0,0 +1,223 @@
+  label.setAttribute("for", id);
+  label.innerHTML = text;
+
+  div.appendChild(label);
+  div.appendChild(radBtn);
+};
+
+const makeTextInput = (id, div, text, ref) => {
+  let messageText = document.createElement("INPUT");
+  messageText.setAttribute("id", id);
+  messageText.setAttribute("type", "text");
+  messageText.setAttribute("value", text);
+  messageText.setAttribute("size", 85);
+
+  if (ref) {
+    div.insertBefore(messageText, ref)
+  } else {
+    div.appendChild(messageText);
+  }
+};
+
+const makeNumInput = (id, div, num, ref) => {
+  let messageProb = document.createElement("INPUT");
+  messageProb.setAttribute("id", id);
+  messageProb.setAttribute("type", "number");
+  messageProb.setAttribute("value", num);
+  messageProb.setAttribute("size", 1);
+
+  if (ref) {
+    div.insertBefore(messageProb, ref)
+  } else {
+    div.appendChild(messageProb);
+  }
+}
+
+let viewItr = 0;
+const views = ['http://localhost:3006/glyphs/5d3744e584687c30ef9e5b47/', 'http://localhost:3006/glyphs/5d37452784687c30ef9e5b48/',
+  'http://localhost:3006/glyphs/5d4885f18227675bc73b98ac/'
+]
+
+views.forEach(view => {
+  $.getJSON(view, function(data) {
+    let viewMsgItr = 0;
+    viewItr++;
+
+    data.view.messages.forEach(element => {
+      viewMsgItr++;
+
+      const currentDiv = document.getElementById(data.name);
+
+      makeTextInput("input-" + viewItr + "-" + viewMsgItr, currentDiv, element.text, null);
+      makeNumInput("prob-" + viewItr + "-" + viewMsgItr + "-1", currentDiv, element.probability, null);
+
+      // Create an update button and set it to post on click
+      makeButton('post-btn' + viewItr + "-" + viewMsgItr, currentDiv, "Update", null);
+
+      $('#post-btn' + viewItr + "-" + viewMsgItr).click(function() {
+        updateView(this);
+      });
+
+      if (data.view.messages.indexOf(element) === data.view.messages.length - 1) {
+        currentDiv.appendChild(document.createElement("BR"));
+        makeButton('add-btn' + viewItr, currentDiv, "Add", null);
+        viewMsgItr++;
+
+        $('#add-btn' + viewItr).click(function() {
+          const id = (this.id.substring(7, 8)) + "-" + (this.id.substring(9, ));
+
+          makeTextInput("input-" + id + viewMsgItr, currentDiv, "", this);
+          makeNumInput("prob-" + id + viewMsgItr + "-" + 1, currentDiv, 0, this);
+          makeButton("post-btn" + id + viewMsgItr, currentDiv, "Update", this);
+
+          document.getElementById("post-btn" + id + viewMsgItr).addEventListener("click", function() {
+            updateView(this);
+          });
+
+          currentDiv.insertBefore(document.createElement("BR"), this);
+        });
+      }
+      currentDiv.appendChild(document.createElement("BR"));
+    });
+    data.view.gauges.forEach(gauge => {
+      let gaugeItr = 0;
+      gaugeItr++;
+      let messageItr = 0;
+
+      currentDiv = document.getElementById(data.name);
+
+      if (typeof(gauge.messages) == 'undefined') {
```

Comment:
I don't know ... I kind of feel like I would like our interface to allow adding a first message to a gauge if it currently contains none.

Reply:
Also, strict equality is needed.

Reply:
This is being worked on, I believe.